state.go

```diff
@@ -266,29 +278,37 @@ func Cmd(cmd ...string) llb.StateOption {
 // AddLabel records a LABEL to the image config.
 func AddLabel(key, value string) llb.StateOption {
 	return func(st llb.State) llb.State {
-		return withImageConfigMutator(st, func(_ context.Context, img *mdispec.DockerOCIImage) error {
+		return withImageConfigMutator(st, func(ctx context.Context, img *ImageConfig) error {
 			if img.Config.Labels == nil {
 				img.Config.Labels = make(map[string]string)
 			}
+			if img.ContainerConfig.Labels == nil {
+				img.ContainerConfig.Labels = make(map[string]string)
+			}
 
 			if curVal, ok := img.Config.Labels[key]; ok && curVal == value {
 				// No need to add the label if it already exists with the same value
 				return nil
 			}
 
 			img.Config.Labels[key] = value
+			img.ContainerConfig.Labels[key] = value // legacy
 
 			// In Dockerfile, multiple labels can be specified in the same LABEL command
 			// leading to one history element. This checks if the previous history
 			// committed was also a label, in which case it should just add to the
 			// previous history element.
+			// The sentinelTime is to ensure we only append to history records
+			// created in this session.
+			sentinelTime := sentinelHistoryTime(ctx)
```

Comment:
I'd suggest factoring this out into a `addOrEditHistoryForKV(img, "LABEL", key, value)`

Reply:
added a commitHistoryKV helper to capture this logic